packages/components/RadioGroup/src/RadioButton.macos.tsx

```diff
@@ -0,0 +1,56 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { ensureNativeComponent } from '@fluentui-react-native/component-cache';
+import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
+import { RadioGroupContext } from './RadioGroup';
+import { radioButtonName, IRadioButtonProps, IRadioButtonSlotProps, IRadioButtonType } from './RadioButton.types';
+
+const NativeRadioButtonView = ensureNativeComponent('RadioButtonView');
+
+export const RadioButton = compose<IRadioButtonType>({
+  displayName: radioButtonName,
+  usePrepareProps: (userProps: IRadioButtonProps, useStyling: IUseComposeStyling<IRadioButtonType>) => {
+    const { content, buttonKey, disabled } = userProps;
+    const info = React.useContext(RadioGroupContext);
+    // Reroute the native component's onPress event to RadioGroup's onChange
+    const onPressRerouted = () => {
+      // Prevent calls to RadioGroup's onChange on the currently selected button
+      if (buttonKey != info.selectedKey) {
+        info.onChange && info.onChange(buttonKey);
+      }
+    };
+
+    const styleProps = useStyling(userProps);
+    const slotProps = mergeSettings<IRadioButtonSlotProps>(styleProps, {
+      root: {
+        buttonKey: buttonKey,
+        content: content,
+        disabled: disabled,
+        onPress: onPressRerouted,
+        selected: info.selectedKey === buttonKey,
+        style: {
+          // Fluent controls are designed to snap to a 4 px grid
+          marginLeft: 4,
+          marginTop: 4,
+          width: 150,
+          height: 20,
+        },
```

Comment:
Do we actually need the margins?
Also 150 / 4 = 37.5, not a 4 px grid? The width is dependent on the label and will probably be overridden by client anyway so it's probably ok..

Reply:
We should probably file a bug on investigating implementing RCTShadowView natively so we don't need to keep hardcoding these...

Reply:
The margins add paddings to the top and left of the radio button, I don't think it's relative to the button width? 
With 4px margins
![image](https://user-images.githubusercontent.com/67026167/129817327-8600628d-e76a-4fb2-a225-427ee8ec3302.png)
Without margins
![image](https://user-images.githubusercontent.com/67026167/129817388-89c27cde-dd6c-43fb-8c8b-87b2c02e9698.png)

Reply:
I filed a bug for the content size issue: https://office.visualstudio.com/APEX/_workitems/edit/5351152